packages/data-service/src/data-service.ts

```diff
@@ -2645,6 +2695,31 @@ class DataServiceImpl extends WithLogContext implements DataService {
     };
   }
 
+  async listStreamProcessors(filter?: Document): Promise<StreamProcessor[]> {
+    const { streamProcessors } = await this._runStreamCmd({
+      listStreamProcessors: 1,
+      filter,
+    });
+    return streamProcessors;
+  }
+
+  async startStreamProcessor(name: string): Promise<void> {
+    await this._runStreamCmd({ startStreamProcessor: name });
+  }
+
+  async stopStreamProcessor(name: string): Promise<void> {
+    await this._runStreamCmd({ stopStreamProcessor: name });
+  }
+
+  async dropStreamProcessor(name: string): Promise<void> {
+    await this._runStreamCmd({ dropStreamProcessor: name });
+  }
+
+  private async _runStreamCmd(cmd: Document, options: Document = {}) {
+    const result = await this._database('admin', 'CRUD').command(cmd, options);
+    return result.ok === 1 ? result : Promise.reject(result);
```

Comment:
I don't think we ever just "throw" raw results from the driver if driver itself doesn't throw. Usually driver will throw if result is not ok. Is is not the case here? If not, can we throw a meaningful error that is constructed as error, and not just driver response?

Reply:
Didn't know driver throws when `ok !== 1`. Will remove this check.